sunpy/map/map_factory.py

```diff
@@ -172,9 +173,16 @@ def _parse_args(self, *args, silence_errors=False, allow_errors=False, **kwargs)
             elif isinstance(arg, str) and is_url(arg):
                 # Replace URL string with a Request object to dispatch on later
                 args[i] = Request(arg)
-            elif possibly_a_path(arg):
+            elif possibly_a_path(arg) and not is_uri(arg):
                 # Replace path strings with Path objects
                 args[i] = pathlib.Path(arg)
+            elif is_uri(arg):
+                # Replace uri string with list of fsspec OpenFile objects
+                if 'fsspec_kwargs' in kwargs:
+                    fsspec_kw = kwargs['fsspec_kwargs']
+                else:
+                    fsspec_kw = {}
+                args[i] = fsspec.open_files(arg, **fsspec_kw)
```

Comment:
Would you mind elaborating why you don't think it will work? It seemed to work as expected while I was testing. This was what I was running for single and multiple files:
```
single_map = sunpy.map.Map("s3://gov-nasa-hdrl-data1/sdo/aia/20100513/0131/sdo_aia_h2_20100513T000400_0131_v1.fits", fsspec_kwargs={'anon':True})
multi_map = sunpy.map.Map("s3://gov-nasa-hdrl-data1/sdo/aia/20100513/0131/sdo_aia_h2_20100513T000*_v1.fits", fsspec_kwargs={'anon':True})
```